### DISCUSSION TITLE
MacOS execute segv, because of Init

### DISCUSSION THREAD

**Discussion by M7T5M3P** [SFML/imgui-sfml#265] [General]
# README: Segmentation Fault Issue on MacOS M3 with ImGui-SFML Integration

## Environment Details
- **Operating System**: MacOS M3
- **Compiler**: set(CMAKE_CXX_STANDARD 17)
- **SFML Version**: 2.6.1
- **ImGui Version**: v1.89.9-docking
- **ImGui-SFML Version**: 2.6.x

## Issue Description
Encountering a segmentation fault when attempting to initialize ImGui with SFML in a MacOS M3 environment. This issue arises specifically during the execution of the `ImGui::SFML::Init(window)` function.

## Steps to Reproduce
1. Initialization of the SFML window in the main function:
   ```cpp
   sf::RenderWindow window;
   window.create(sf::VideoMode(1920, 1080), "ImGui + SFML = <3");
   window.setFramerateLimit(60);

   if (!ImGui::SFML::Init(window)) {
       std::cerr << "ImGui::SFML::Init(window) failed" << std::endl;
       return 84;
   }
   ```
   This setup is based on the guidelines provided in the ImGui-SFML README.md.

2. The project is built using CMake, ensuring the correct fetching and use of specified versions of SFML, ImGui, and ImGui-SFML.

## Additional Observations
- In the ImGui-SFML source code, the `Init` function seems to require two parameters. This is inconsistent with the provided example where only the SFML window is passed.
- The CMake configuration is set to fetch specific versions of the necessary libraries.

## Request for Assistance
I am seeking:
- Suggestions or solutions to resolve the segmentation fault.
- Clarification on the proper usage of the `ImGui::SFML::Init` function in this context.

Any help or guidance from the community to troubleshoot this issue would be appreciated. Thanks :)

**Comment by M7T5M3P** [comment]
Okay I found the problem it's the call to the demo, function, is it really still working ? Without it, it works perfectly!

**Comment by eliasdaler** [comment]
> In the ImGui-SFML source code, the Init function seems to require two parameters. This is inconsistent with the provided example where only the SFML window is passed.

One of the overloads looks like this:
```cpp
IMGUI_SFML_NODISCARD IMGUI_SFML_API bool Init(sf::RenderWindow& window,
                                              bool loadDefaultFont = true);
```
The second argument has a default value so it's optional.

Please provide the stack trace where exactly it crashes (which line, what variable it tries to access etc.) - it's impossible to tell what goes wrong otherwise.
